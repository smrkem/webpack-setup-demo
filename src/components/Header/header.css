header {
  background: linear-gradient(to right, #53384c, #3f1429);
}

.header-container {
  display: flex;
  justify-content: space-between;
}

header h1 {
  color: #fff;
  font-size: 3.5rem;
  text-transform: uppercase;
}

header nav ul {
  height: 100%;
  display: flex;
  align-items: center;
}

header nav a {
  color: #fff;
  text-transform: uppercase;
  font-size: 2rem;
  margin-left: 2rem;
}

header nav a:focus,
header nav a:hover {
  color: #990066;
}
