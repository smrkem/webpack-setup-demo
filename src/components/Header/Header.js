import React from 'react'
import { Link } from 'react-router-dom'

import './header.css'

const Header = () => (
  <header>
    <div className="container header-container">
      <h1>Matt Smrke</h1>
      <nav>
        <ul>
          <li><Link to='/'>Posts</Link></li>
          <li><Link to='/about'>About</Link></li>
        </ul>
      </nav>
    </div>
  </header>
)

export default Header
