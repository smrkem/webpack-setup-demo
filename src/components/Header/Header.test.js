import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from './Header'

test('Header should render as expected', () => {
  const component = shallow(<Header />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
