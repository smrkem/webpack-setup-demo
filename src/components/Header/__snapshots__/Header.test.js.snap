// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render as expected 1`] = `
<header>
  <div
    className="container header-container"
  >
    <h1>
      Matt Smrke
    </h1>
    <nav>
      <ul>
        <li>
          <Link
            replace={false}
            to="/"
          >
            Posts
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/about"
          >
            About
          </Link>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
